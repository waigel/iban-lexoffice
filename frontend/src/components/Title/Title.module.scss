.colorfull {
  background: -webkit-linear-gradient(
    45deg,
    #d93ea5 18.93%,
    #f37653 38.38%,
    #fbbc05 71.6%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: inline-block;
}

.size {
  font-size: 4rem;
  height: auto;
}
