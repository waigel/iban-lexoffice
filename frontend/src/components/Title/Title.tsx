import React from 'react';
import {classNames} from '../../utils/classNames';
import styles from './Title.module.scss';

export const Title = (props: React.HtmlHTMLAttributes<HTMLHeadingElement>) => {
  return (
    <h1
      {...props}
      className={classNames(
        props.className,
        'font-omesbold tracking-normal',
        styles.colorfull,
        styles.size
      )}
    />
  );
};
