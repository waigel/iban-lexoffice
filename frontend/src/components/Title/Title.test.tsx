import React from 'react';
import TestRenderer from 'react-test-renderer';
import {Title} from './Title';

it('should test the Title component is rendered correctly', () => {
  const tree = TestRenderer.create(<Title>TEST TITLE</Title>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should test the Title component is rendered correctly with additional className', () => {
  const tree = TestRenderer.create(
    <Title className={'test2'}>TEST TITLE</Title>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
