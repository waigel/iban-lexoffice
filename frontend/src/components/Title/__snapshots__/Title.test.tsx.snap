// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Title component is rendered correctly 1`] = `
<h1
  className="font-omesbold tracking-normal colorfull size"
>
  TEST TITLE
</h1>
`;

exports[`should test the Title component is rendered correctly with additional className 1`] = `
<h1
  className="test2 font-omesbold tracking-normal colorfull size"
>
  TEST TITLE
</h1>
`;
